### PR TITLE
New header and stuff

### PR DESCRIPTION
**Pull request by yoggsoft**
* General
  * Add homepage styles
  * Restrict button colour contrast
  * Add pre-footer styles
  * Split functions and mixins
  * Replace modernizr behavior
  * Add user icon
* Navigation bar
  * Fix navbar-btn
  * Improve behavior on mobile menu

### CODE REVIEW

File: src/sass/colors/_colors-helpers.scss [hunk 1]
```diff
@@ -32,9 +32,32 @@ Styleguide Colors.Helpers
   .background-#{$color} {
     background-color: $colorValue;
   }
-  .btn-#{$color} {
-    color: $white;
-    background-color: $colorValue;
+  .btn-#{$color} { // Respect button contrast
+    @if (str-index($color, 'white') or str-index($color, 'light') or str-index($color, 'lighter') or str-index($color, 'smoke') ) {
+      color: $grey-darker !important;
```
Comment (mklinga) [inline src/sass/colors/_colors-helpers.scss]: Is there a chance we could move this after the regular button stylings on that file so we could get rid of `!important`s?

File: src/sass/patterns/_navbar.scss [hunk 2]
```diff
@@ -61,87 +90,112 @@ Markup:
...
+  }
+}
+
+.overlay {
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  width: 100%;
+  z-index: 1000;
```
Comment (mklinga) [inline src/sass/patterns/_navbar.scss]: Maybe use our `$zindex-xyz` variables here?

File: src/sass/patterns/_navbar.scss [hunk 3]
```diff
@@ -151,82 +205,59 @@ Styleguide Patterns.Navbar
...
 
 .navbar-toggle,
 .navbar-toggle-like {
-  border: none;
-  background: none;
-  margin: 0;
-  line-height: 0;
-  padding: 10px 15px;
   position: absolute;
+  float: left;
```
Comment (mklinga) [inline src/sass/patterns/_navbar.scss]: I don't think `float: left` does anything with `position: absolute` set

File: src/sass/colors/_colors-helpers.scss (hunk 1 above)
Comment (yoggsoft) [inline src/sass/colors/_colors-helpers.scss]: In this case, `!important` is intended so button color contrast by definition is always respected 
- Darker buttons will always have white icons and font
- Lighter buttons will always have black(grey-darker) fonts
So this means, If I add a `btn btn-warning` inside any structure, (navbar, settings menu, floating modal, pricing box) it's contrast will be respected, so no we never have to go custom buttons.
This is to ensure e.g.`btn btn-primary` doesn't get black font/icons on any scenario and avoid stuff like:
`body.settings { 
   &.btn.btn-primary {
      color: white;
      &.icon{
          fill: white;
      }
   }
}`

**Pull request merged**